cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -0,0 +1,351 @@
+                promise.fail(new RuntimeException("Failed to update Zookeeper server configuration", e));
+            }
+        }, false, configPromise);
+
+        return configPromise.future();
+    }
+
+    /**
+     * Closes the Zookeeper connection asynchronously
+     *
+     * @return
+     */
+    private Future<Void> closeConnectionAsync() {
+        Promise<Void> closePromise = Promise.promise();
+
+        log.debug("Closing Zookeeper connection to {}", zookeeperConnectionString);
+
+        vertx.executeBlocking(promise -> {
+            closeConnectionSync();
+            promise.complete();
+        }, true, closePromise);
+
+        return closePromise.future();
+    }
+
+    /**
+     * Closes the Zookeeper connection synchronously
+     */
+    private void closeConnectionSync() {
+        if (zkAdmin != null)    {
+            try {
+                zkAdmin.close();
+            } catch (InterruptedException e) {
+                log.warn("Failed to close the ZooKeeperAdmin", e);
+            } finally {
+                zkAdmin = null;
+            }
+        }
+    }
+
+    /**
+     * Generates the TLS configuration for Zookeeper.
+     *
+     * @return
+     */
+    private ZKClientConfig getClientConfig()  {
+        ZKClientConfig clientConfig = new ZKClientConfig();
+
+        clientConfig.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty");
+        clientConfig.setProperty("zookeeper.client.secure", "true");
+        clientConfig.setProperty("zookeeper.ssl.trustStore.location", trustStoreFile.getAbsolutePath());
+        clientConfig.setProperty("zookeeper.ssl.trustStore.password", trustStorePassword);
+        clientConfig.setProperty("zookeeper.ssl.trustStore.type", "PKCS12");
+        clientConfig.setProperty("zookeeper.ssl.keyStore.location", keyStoreFile.getAbsolutePath());
+        clientConfig.setProperty("zookeeper.ssl.keyStore.password", keyStorePassword);
+        clientConfig.setProperty("zookeeper.ssl.keyStore.type", "PKCS12");
+
+        return clientConfig;
+    }
+
+    /**
+     * Converts the map with configuration to List of Strings which is the format in which the ZookeeperAdmin client
+     * expects the new configuration.
+     *
+     * @param servers   Map with Zookeeper configuration
+     * @return          List with Zookeeper configuration
+     */
+    /*test*/ static List<String> serversMapToList(Map<String, String> servers)  {
+        List<String> serversList = new ArrayList<String>(servers.size());
+
+        for (Map.Entry entry : servers.entrySet())  {
+            serversList.add(String.format("%s=%s", entry.getKey(), entry.getValue()));
+        }
+
+        return serversList;
+    }
+
+    /**
+     * Parse the byte array we get from Zookeeper into a map we use internally. During the parsing, only server
+     * configurations are used. The version field is ignored.
```

Comment:
It wasn't clear to me what "During the parsing, only server configurations are used." meany until I read the code.